### ISSUE TITLE
Docker entrypoint script optimization for external etcd usage

### ISSUE THREAD

**Issue by rgarcia89** [sensu/sensu-go#4147]
The current docker entrypoint script is not covering the case that an external etcd is used.
This is causing a lot of the following log message, which could be misleading

```
echo "== waiting for ${SENSU_HOSTNAME}:${WAIT_PORT} to become available before running backend-init..."
```

I would like to request an optimization of that script.
Basically every etcd related step within that script can be seen as obsolet, if an external etcd is used.
What would make sense is to remove all etcd related variables and set the etcd availability check to use the url provided in `--etcd-client-urls=`

**Comment by sensu-discourse** [comment]
This issue has been mentioned on **Sensu Community**. There might be relevant details there:

https://discourse.sensu.io/t/best-practices-update-sensu-backend-with-zero-downtime/1821/51

**Comment by taasest8** [comment]
@sensu-discourse we are also experiencing the same issue as @rgarcia89 and the link you posted to discourse leads to a workaround that was suggested by @rgarcia89 who opened this issue. 

But as he mentioned, it is only a workaround and the scripted should be adjusted for installations with a remote etcd database backend as this is a supported scenario.

**Comment by tgdfool2** [comment]
Just tested the workaround in our infrastructure (using an external `etcd` cluster) and I can confirm that it does the trick.

I would be happy to help adjusting the entrypoint script, but unfortunately it seems that the source code is not publicly available:
* https://github.com/sensu/sensu-enterprise-go

(At least I receive a 404)

**Comment by calebhailey** [comment]
@rgarcia89 (et al) the new `sensu-backend init --wait` flag will ship in version 6.3.0 (current ETA is early Feb), which should resolve this issue.

**Comment by rgarcia89** [comment]
@calebhailey I am not sure how this should help with my issue.
Could you please review what I wrote https://github.com/sensu/sensu-go/issues/4147#issue-774910546
It is not about waiting - when using an external etcd it is more about making the check that is build in know about that.

**Comment by amdprophet** [comment]
@rgarcia89 adding support for waiting in the `sensu-backend init` command allows us to remove the wait logic (including the `== waiting for` message) from `entrypoint.sh`. This will be shipped with 6.3.0.

The other etcd variables are currently necessary for the image to run in certain scenarios, using internal etcd, without manually setting those variables. I suspect this is due to another issue where we override the etcd defaults in our command parsing. This will be addressed by https://github.com/sensu/sensu-go/issues/3952.
